cluster-autoscaler/cloudprovider/clusterapi/clusterapi_converters.go

```diff
@@ -42,6 +42,22 @@ func newMachineDeploymentFromUnstructured(u *unstructured.Unstructured) *Machine
 		Status: MachineDeploymentStatus{},
 	}
 
+	machineSpec, found, err := unstructured.NestedMap(u.Object, "spec", "template", "spec")
```

Comment:
Just thinking, there's an awful lot of duplication in this file, might be good to refactor at some point so we have things like `newMachineSpecFromUnstructured` that can be reused within the machine, machineset and machinedeployment converters 🤔

Reply:
++, i will definitely clean this up

Reply:
i cleaned it up some, let me know if this it's looking better.